assignment/assignment-impl/impl/src/java/org/sakaiproject/assignment/impl/BaseAssignmentService.java

```diff
@@ -11742,6 +11707,62 @@ else if (typeOfGrade == Assignment.CHECK_GRADE_TYPE) {
 				}
 			}
 		}
+		
+		
+		public String getGradeForUserInGradeBook(String userId)
+		{
+			return this.getGradeFromGradebook(userId);
+		}
+		
+		protected String getGradeFromGradebook(String userId)
+		{
+			String rv =null;
+			if (userId == null) 
+			{
+				userId = m_submitterId;
+			}
+			Assignment m = getAssignment();
+			String gAssignmentName = StringUtils.trimToNull(m.getProperties().getProperty(AssignmentService.PROP_ASSIGNMENT_ASSOCIATE_GRADEBOOK_ASSIGNMENT));
+			if (gAssignmentName != null)
+			{
+				GradebookService g = (GradebookService)  ComponentManager.get("org.sakaiproject.service.gradebook.GradebookService");
+				String gradebookUid = m.getContext();
+
+				// return student score from Gradebook
```

Comment:
You can replace almost everything from line 11731 to 11765 with this one line:

final GradeDefinition def = this.gradebookService.getGradeDefinitionForStudentForItem(gradebookUid,
					assignment.getId(), studentUuid);

Note that you'll need the assignment id, look it up via GradebookService.getAssignment. Though deprecated, it won't go away since Assignments (wrongly) uses names instead of ids.